p2p/src/net/mock/peer.rs

```diff
@@ -60,6 +61,13 @@ pub struct Peer<T: TransportSocket> {
 
     /// RX channel for receiving commands from backend
     rx: mpsc::Receiver<MockEvent>,
+
+    /// List of nonces of all active connections
+    handshake_nonce_store: Arc<HandshakeNonceStore>,
```

Comment:
I'm very confused to why this even exists. Why do we have to store a list of all nonces from all peers?

The only purpose of this nonce is to detect if we're connecting to ourselves somehow, which can never be verified with networking means because proxies exist. So, we have a very simple tool to make that check.

Now I see we're storing the nonce of every single peer. What could we possibly use that for? We cannot use it to id the peer because a malicious peer can just create an infinite number of them arbitrarily... these numbers aren't authenticated by any means. We're back at the "peer id" thing, I feel.

Maybe I'm missing something... I don't know.

Reply:
Sorry, the comment was wrong. I updated it:

```
    /// List of nonces of all outbound connections in handshake state.
    /// Used to detect and drop self connections.
    handshake_nonce_store: Arc<HandshakeNonceStore>,
```

I also now remove own nonce from the store once handshake complete, because we don't need it after that.